packages/medusa/src/services/__tests__/event-bus.js

```diff
@@ -102,26 +85,100 @@ describe("EventBusService", () => {
     describe("successfully adds job to queue", () => {
       beforeAll(() => {
         jest.resetAllMocks()
-        const stagedJobRepository = MockRepository({
-          find: () => Promise.resolve([]),
-        })
 
         eventBus = new EventBusService({
           logger: loggerMock,
           manager: MockManager,
-          stagedJobRepository,
         })
 
         eventBus.queue_.add.mockImplementationOnce(() => "hi")
 
         job = eventBus.emit("eventName", { hi: "1234" })
       })
+
+      it("calls queue.add", () => {
+        expect(eventBus.queue_.add).toHaveBeenCalled()
+      })
+    })
+
+    describe("successfully adds job to cache if cache key is provided", () => {
+      const cacheService = new CacheServiceMock()
+      beforeAll(() => {
+        jest.resetAllMocks()
+
+        eventBus = new EventBusService({
+          logger: loggerMock,
+          manager: MockManager,
+          cacheService,
+        })
+
+        job = eventBus.emit("eventName", { hi: "1234" }, { cacheKey: "test" })
+      })
       afterAll(async () => {
         await eventBus.stopEnqueuer()
       })
 
-      it("calls queue.add", () => {
-        expect(eventBus.queue_.add).toHaveBeenCalled()
+      it("calls cacheservice", () => {
+        expect(cacheService.get).toHaveBeenCalledTimes(1)
+        expect(cacheService.set).toHaveBeenCalledTimes(1)
+      })
+
+      it("does not add to queue", () => {
+        expect(eventBus.queue_.add).toHaveBeenCalledTimes(0)
+      })
+    })
+  })
+
+  describe("processCachedEvents", () => {
+    let eventBus
+    let job
+
+    const cacheService = {
+      get: (cacheKey) => {
+        if (cacheKey === "test") {
```

Comment:
suggestion: everywhere we are depending on some ids to be then used later or in another place, I would suggest storing it in a const as it can be annoying to try to find an id without knowing where it is located. Also, if we change the id for any reason it is then centralized and much easier. It is more of a global things but that can be applied here